lib/ui/text.dart

```diff
@@ -80,15 +81,15 @@ class FontWeight {
   ///
   /// Values for `t` are usually obtained from an [Animation<double>], such as
   /// an [AnimationController].
-  static FontWeight/*?*/ lerp(FontWeight/*?*/ a, FontWeight/*?*/ b, double/*!*/ t) {
-    assert(t != null);
+  static FontWeight? lerp(FontWeight? a, FontWeight? b, double t) {
+    assert(t != null); // ignore: unnecessary_null_comparison
     if (a == null && b == null)
       return null;
-    return values[lerpDouble(a?.index ?? normal.index, b?.index ?? normal.index, t).round().clamp(0, 8) as int];
+    return values[_lerpInt(a?.index ?? normal.index, b?.index ?? normal.index, t).round().clamp(0, 8) as int];
```

Comment:
Drive by comment: `a?.index ?? normal.index` could be written as `(a ?? normal).index`. It's shorter and *(IMO) more readable.

Reply:
Neat! Done.